plugins/kubernetes/app/models/watchers/deploy_watcher.rb

```diff
@@ -7,113 +6,211 @@ class DeployWatcher
 
     finalizer :on_termination
 
-    def initialize(release)
-      @release = release
-      @current_rcs = {}
+    def initialize(project)
+      @project = project
       @pod_timer = after(ENV.fetch('KUBERNETES_POD_TIMEOUT', 10.minutes).to_i) { pod_timeout }
-      info "Start watching K8s deploy: #{@release}"
+      start_watching
+    end
+
+    def start_watching
+      sync_with_cluster
+      info 'Start watching for deployments'
       async :watch
     end
 
     def watch
-      subscribe("#{@release.build.project_name}", :handle_update)
-    end
-
-    def handle_update(topic, data)
-      release_doc = release_doc_from_rc_name(topic)
-      release_updated = if data.object.kind == 'Event'
-                        handle_event_update(release_doc)
-                      else
-                        handle_pod_update(release_doc, data)
-                      end
-      if release_updated
-        send_event(role: release_doc.kubernetes_role.name,
-                   deploy_group: release_doc.deploy_group.name,
-                   target_replicas: release_doc.replica_target,
-                   live_replicas: release_doc.replicas_live,
-                   failed: release_doc.failed?)
-        end_deploy if deploy_finished?
+      subscribe("pod-events-project-#{@project.id}", :handle_event)
```

Comment:
It would be good to have the topic constructed in a single place. Right now it's built both here and in the pod watcher.